src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
-
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle sides');
+    }
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = Shape.Circle;
+
+  color: Color;
 
+  radius: number;
+
+  constructor(
+    color: Color,
+    radius: number,
```

Comment:
Consider rewriting this as in the previous Figure
```suggestion
    public radius: number,
```